stackslib/src/chainstate/burn/sortition.rs

```diff
@@ -604,4 +923,250 @@ mod test {
+            5
+        ));
+        assert!(BlockSnapshot::check_miner_is_active(
+            StacksEpochId::Epoch30,
+            1,
+            &BurnchainSigner("".to_string()),
+            4
+        ));
+        assert!(BlockSnapshot::check_miner_is_active(
+            StacksEpochId::Epoch30,
+            1,
+            &BurnchainSigner("".to_string()),
+            3
+        ));
+        assert!(BlockSnapshot::check_miner_is_active(
+            StacksEpochId::Epoch30,
+            1,
+            &BurnchainSigner("".to_string()),
+            2
+        ));
+        assert!(BlockSnapshot::check_miner_is_active(
+            StacksEpochId::Epoch30,
+            1,
+            &BurnchainSigner("".to_string()),
+            1
+        ));
+        assert!(!BlockSnapshot::check_miner_is_active(
+            StacksEpochId::Epoch30,
+            1,
+            &BurnchainSigner("".to_string()),
+            0
+        ));
+    }
+
+    #[test]
+    fn test_get_miner_commit_carryover() {
+        assert_eq!(
+            BlockSnapshot::get_miner_commit_carryover(None, None),
+            (AtcRational::zero(), false)
+        );
+        assert_eq!(
+            BlockSnapshot::get_miner_commit_carryover(None, Some(1)),
+            (AtcRational::zero(), false)
+        );
+        assert_eq!(
+            BlockSnapshot::get_miner_commit_carryover(Some(1), None),
+            (AtcRational::zero(), false)
+        );
+
+        // ATC increased
+        assert_eq!(
+            BlockSnapshot::get_miner_commit_carryover(Some(1), Some(1)),
+            (AtcRational::one(), false)
+        );
+        assert_eq!(
+            BlockSnapshot::get_miner_commit_carryover(Some(2), Some(1)),
+            (AtcRational::one(), false)
+        );
+
+        // no carried commit
+        assert_eq!(
+            BlockSnapshot::get_miner_commit_carryover(Some(2), Some(0)),
+            (AtcRational::zero(), true)
+        );
+
+        // assumed carryover
+        assert_eq!(
+            BlockSnapshot::get_miner_commit_carryover(Some(2), Some(4)),
+            (AtcRational::frac(2, 4), true)
+        );
+    }
+
+    #[test]
+    fn test_null_miner_logistic() {
+        for i in 0..1024 {
+            let atc_u256 = ATC_LOOKUP[i];
+            let null_miner_lgst =
+                BlockSnapshot::null_miner_logistic(AtcRational::frac(i as u64, 1024));
+            assert_eq!(null_miner_lgst, atc_u256);
+        }
+        assert_eq!(
+            BlockSnapshot::null_miner_logistic(AtcRational::zero()),
+            ATC_LOOKUP[0]
+        );
+        assert_eq!(
+            BlockSnapshot::null_miner_logistic(AtcRational::one()),
+            *ATC_LOOKUP.last().as_ref().cloned().unwrap()
+        );
+        assert_eq!(
+            BlockSnapshot::null_miner_logistic(AtcRational::frac(100, 1)),
+            *ATC_LOOKUP.last().as_ref().cloned().unwrap()
+        );
+    }
+
+    #[test]
+    fn test_null_miner_wins() {
```

Comment:
Can you add a comment to this function that describes what it is testing and how it tests it?

Reply:
Sure